gnome-image-installer/pages/install/gis-install-page.c

```diff
@@ -71,6 +74,58 @@ G_DEFINE_QUARK(install-error, gis_install_error);
 #define IMAGE_KEYRING "/usr/share/keyrings/eos-image-keyring.gpg"
 
 static gboolean
+delete_event_cb (GtkWidget      *toplevel,
+                 GdkEvent       *event,
+                 GisInstallPage *self)
+{
+  GisInstallPagePrivate *priv = gis_install_page_get_instance_private (self);
+  GtkWidget *button;
+  gboolean should_propagate;
+  gint response_id;
+
+  /* If we're not writing, it's still safe to quit */
+  if (!priv->writing)
+    return GDK_EVENT_PROPAGATE;
+
+  priv->warning_dialog = gtk_message_dialog_new (GTK_WINDOW (toplevel),
+                                                 GTK_DIALOG_MODAL | GTK_DIALOG_DESTROY_WITH_PARENT,
+                                                 GTK_MESSAGE_WARNING,
+                                                 GTK_BUTTONS_NONE,
+                                                 _("Abort reformatting the disk?"));
```

Comment:
I think the verb should be consistent throughout the message / secondary text / button, but at the moment it's "abort" / "cancelling" / "stop". I think "Stop" is the clearest.